src/client.cc

```
@@ -160,34 +160,57 @@ void ListObjectsResult::Populate() {
+      std::make_shared<std::future<ListObjectsResponse>>(std::async(
+          std::launch::async,
+          [this, next_args = std::move(next_args)]() mutable {
+            std::string region;
+            if (GetRegionResponse resp =
+                    client_->GetRegion(next_args.bucket, next_args.region)) {
+              region = resp.region;
+              if (next_args.recursive) {
+                next_args.delimiter = "";
+              } else if (next_args.delimiter.empty()) {
+                next_args.delimiter = "/";
+              }
```

Cppcheck (2.21.0)

[style] The function 'QuoteFieldsToString' is never used.

(unusedFunction)